### PR TITLE
Fix issue #4625

### PR DESCRIPTION
**Pull request by lnkuiper**
For details, see issue #4625

### CODE REVIEW

**Review by hawkfish — APPROVED** [review]

File: src/common/sort/sort_state.cpp [hunk 1]
```diff
@@ -123,6 +123,32 @@ SortLayout::SortLayout(const vector<BoundOrderByNode> &orders)
...
+	SortLayout result;
+	result.column_count = num_prefix_cols;
+	result.all_constant = true;
+	result.comparison_size = 0;
+	for (idx_t col_idx = 0; col_idx < num_prefix_cols; col_idx++) {
+		result.order_types.push_back(order_types[col_idx]);
+		result.order_by_null_types.push_back(order_by_null_types[col_idx]);
+		result.logical_types.push_back(logical_types[col_idx]);
+
+		result.all_constant = all_constant && constant_size[col_idx];
```
Comment (Tishj) [inline src/common/sort/sort_state.cpp:136]: Won't this flip to true as long as `this->all_constant` is true and the last checked column is true?

So other columns can not be constant and `result.all_constant` would end up being true

**Review by Tishj — COMMENTED** [review]
Just something that caught my eye, I might be seeing this wrong, figured I'd mention it anyways

File: src/common/sort/sort_state.cpp (hunk 1 above)
Comment (hawkfish) [reply src/common/sort/sort_state.cpp:136]: Yes that is actually the point. We don't need to compare heap data in the prefix if there is none.

Comment (lnkuiper) [reply src/common/sort/sort_state.cpp:136]: It cannot flip to true. It starts at true and stays true as long as all columns it passes are constant.

If it encounters a column that is not constant, it flips to false, and stays false.

Comment (lnkuiper) [reply src/common/sort/sort_state.cpp:136]: Ah wait, I see what you mean now ... we're missing a `this`?

**Pull request merged**